_posts/2017-11-17-falling-into-machine-learning.md

```diff
@@ -0,0 +1,62 @@
+---
+author: Sergio Rodrigo Royo
+date: 2017-11-21 08:00:00 +00:00
+layout: post
+asset-type: post
+slug: falling-into-machine-learning
+title: Falling into Machine Learning
+description: Discovering Machine Learning
+canonical:
+    name: my personal blog
+    href: http://srodrigo.me/falling-into-machine-learning/
+image:
+    src: /assets/img/custom/blog/2017-11-21-falling-into-machine-learning.png
+    attribution:
+      text: by Alex Knight
+      href: https://unsplash.com/photos/2EJCSULRwC8
+tags:
+- machine learning
+---
+
+Recently, I rediscovered my passion for Mathematics and Artificial Intelligence, which I used to hate during my degree in Computer Sciences. Lately, I’ve been focused on Software Design, Automated Testing, Microservices, and Functional Programming. I also love learning new programming languages, so I’ve been wondering whether to go deeper into Golang, Scala or Python. But, in the end, all of them are tools; tools for building what kind of things? In my case, my day-to-day job consists in building and maintaining Microservices, as a full-stack developer. I like it, it’s challenging and you need to have many different skills. But, at some point, I started to look for something completely different to learn in my spare time.
+
+One day, I ended up reading about Golang and Python. I noticed someone was mentioning Machine Learning in both languages, and I got curious and started reading about it. After a while, I realised that those two subjects (Mathematics and Artificial Intelligence) that I didn’t quite like and didn’t mean much to me separately, could be combined to create cool (and useful) things in a completely different area than I’m used to. Completely new challenges! Since then, I got excited about Machine Learning and it’s been the most enjoyable and exciting use of my skills in a long time. The tools I used to like learning (mainly programming languages) became secondary, or what they really are: a means to an end. I stopped googling around, and started an online course about Machine Learning. The fun began.
+
+## What can I do with Machine Learning?
+Machine Learning can sound like a mysterious thing, but we actually see it every day. The amount of applications is countless:
+
+* How do Amazon or Netflix know what you like, to make recommendations?
+* What’s behind those self-driving cars?
+* Is it possible to automate financial trading?
+* What about weather forecasts?
+* How does the Google Search Engine work?
+* Can machines help us improve healthcare?
+* What made Siri or Alexa be able to process natural language?
```

Comment:
Might read better if it was . "What gives Siri and Alexa the ability to process natural language?"

Reply:
It does! I struggled to express that sentence.